### PR TITLE
Print pkgpanda exceptions to stderr

### PR DESCRIPTION
**Pull request by amitaekbote**
## High Level Description

Currently pkgpanda exceptions go to stdout. This resulted in upgrade script not printing errors. Changing print in pkgpanda to go to stderr in case of exceptions. 

## Related Issues
https://jira.mesosphere.com/browse/DCOS-15302

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label Work in progress

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label Ready for review

**Comment by amitaekbote** [comment]
@branden @mellenburg ptal :)

**Comment by branden** [comment]
Changes LGTM. Did you test that the upgrade script prints an error when pkgpanda fails?

**Comment by amitaekbote** [comment]
@branden tested manually

@spahl should we backport this to 1.9.1?

**Comment by amitaekbote** [comment]
core@ip-10-0-4-35 ~ $ dcos-shell pkgpanda activate foo
Validation Error: Invalid package id foo. Package ids may only contain one '--' which seperates the name and version

**Comment by spahl** [comment]
Yes. I think we should bacllkport this one.

**Comment by amitaekbote** [comment]
Current master

core@ip-10-0-6-172 ~ $ sudo bash dcos_node_upgrade.sh 
Upgrading DC/OS master 1.9-dev -> 1.9-dev
ERROR: Upgrade failed at line 80
core@ip-10-0-6-172 ~ $  


After fix:

core@ip-10-0-4-35 ~ $ sudo bash dcos_node_upgrade.sh 
Upgrading DC/OS master 1.9-dev -> 1.9-dev
ERROR: Unable to fetch package dcos-config--setup_a411bc7d63a7d73166053571aa0d27403d4791fa: Problem fetching file:///opt/dcos_install_tmp/packages/dcos-config/dcos-config--setup_a411bc7d63a7d73166053571aa0d27403d4791fa.tar.xz to /tmp/tmp3m740luk.tar.xz because of [Errno 2] No such file or directory: '/opt/dcos_install_tmp/packages/dcos-config/dcos-config--setup_a411bc7d63a7d73166053571aa0d27403d4791fa.tar.xz'. Unable to remove partial download. Future builds may have problems because of it.
ERROR: Upgrade failed at line 80

**Comment by amitaekbote** [comment]
@spahl can you review this? ill create a pr against 1.9.1 too

File: pkgpanda/cli.py [hunk 1]
```diff
@@ -210,16 +210,16 @@ def main():
             # Run all checks
             sys.exit(run_checks(checks, install, repository))
     except ValidationError as ex:
```
Comment (spahl) [inline pkgpanda/cli.py]: This breakout into 3 possible informational messages is not really useful in practice IMHO

Comment (branden) [inline pkgpanda/cli.py]: Yeah, at this level we should just have a single handler for expected exceptions and defer to their error messages. IMO the `Exception` handler should go away and unexpected exceptions should bubble up to the top, tracebacks and all.

**Comment by adam-mesos** [comment]
Merged to 1.9.1 branch in #1537 , but still needs to land in master in #1535

**Pull request merged**